Adding ingres-class annotation docs (#4264)
knative/docs: docs/developer/serving/services/ingress-class.md

> # Configuring Services custom Ingress Class
> By default, when a Knative Service is created an Ingress Class (`ingress.class`) is automatically chosen for it based on the content of the `networking-config` ConfigMap located inside the `knative-serving` namespace. 

Reviewer: ```suggestion
# Configuring Services custom ingress class

By default, when a Knative Service is created, an ingress class (`ingress.class`) is automatically assigned to it based on the content of the `networking-config` ConfigMap. The `networking-config` ConfigMap is located inside the `knative-serving` namespace. 
```

slight editing suggestions for readability

Author: done

> The `networking-config` ConfigMap is configured at installation time and by default if no other implementation is provided this is defaulted to `"istio.ingress.networking.knative.dev"`. Note that it is best to only update this value during the setup of Knative, to avoid getting undefined behavior.

Reviewer: ```suggestion
The `networking-config` ConfigMap is configured when Knative Serving is installed, and if no other ingress class is specified, this defaults to `"istio.ingress.networking.knative.dev"`.
```
I'd move the note to an actual note 🙂

Author: done

> Once configuraded in the `networking-config` ConfigMap, the `ingress.class` will be used for all Knative Services which doesn't specify the `ingress.class` annotation.

Reviewer: ```suggestion
After it is configured in the `networking-config` ConfigMap, the `ingress.class` is used for all Knative Services that do not specify an `ingress.class` annotation.

!!! warning
    The ingress class value should only be modified during Knative Serving installation to avoid issues in your deployment.
```

Author: done

> ## Using the `ingress-class` annotation 

Reviewer: ```suggestion
## Using the ingress class annotation 
```
see https://knative.dev/docs/help/contributor/style-guide/style-and-formatting/#do-not-use-code-formatting-inside-headings

Author: done

> Most of the times, you want your Knative Services to use the default and same `ingress.class`, but for scenarios where there are multiple networking implementations you might want to specify different `ingress.class`es per service. 

Reviewer: ```suggestion
Generally it is recommended for Knative Services to use the default `ingress.class`. However, in scenarios where there are multiple networking implementations, you might want to specify different ingress class annotations for each Service. 
```

Author: done

> You can configure each Service to use a different Ingress Class by specifying the `network.knative.dev/ingress.class` label.

Author: done, yeah.. that was a silly mistake sorry about that

> - To label a Knative Service:
> 
>     ```bash
>     kubectl annotate kservice ${KSVC_NAME} network.knative.dev/ingress.class=<Implementation>
>     ```
> 
> This annotation overrides the configuration specified in the `networking-config` ConfigMap.

Reviewer: ```suggestion
To add an ingress class annotation to a Service, run the following command:

    ```bash
    kubectl annotate kservice <service-name> network.knative.dev/ingress.class=<ingress-type>
    ```
    Where:
    - `<service-name>` is the name of the Service that you are applying the annotation to.
    - `<ingress-type` is the type of ingress that is used as the ingress class for the Service.

!!! note
    This annotation overrides the configuration specified in the `networking-config` ConfigMap.
```

For `<ingress-type>` here, considering mentioning what the possible values are besides `"istio.ingress.networking.knative.dev"`?

Author: @abrennan89 this is an extension point so I didn't want to list the current implementations only. If you don't have a particular implementation installed it doesn't make sense to say that is available here.

Reviewer: ACK, thanks

The `networking-config` ConfigMap is configured when Knative Serving is installed, and if no other ingress class is specified, this defaults to `"istio.ingress.networking.knative.dev"`. 

After it is configured in the `networking-config` ConfigMap, the `ingress.class` is used for all Knative Services that do not specify an `ingress.class` annotation.

!!! warning
    The ingress class value should only be modified during Knative Serving installation to avoid issues in your deployment.

## Using the ingress class annotation

Generally it is recommended for Knative Services to use the default `ingress.class`. However, in scenarios where there are multiple networking implementations, you might want to specify different ingress class annotations for each Service.

You can configure each Service to use a different ingress class by specifying the `network.knative.dev/ingress.class` annotation.

To add an ingress class annotation to a Service, run the following command:
```bash
kubectl annotate kservice <service-name> network.knative.dev/ingress.class=<ingress-type>
```

> Where:
> - `<service-name>` is the name of the Service that you are applying the annotation to.
> - `<ingress-type` is the type of ingress that is used as the ingress class for the Service.

Reviewer: ```suggestion
Where:

- `<service-name>` is the name of the Service that you are applying the annotation to.
- `<ingress-type` is the type of ingress that is used as the ingress class for the Service.
```

>     Where:
>     - `<service-name>` is the name of the Service that you are applying the annotation to.
>     - `<ingress-type` is the type of ingress that is used as the ingress class for the Service.

Reviewer: ```suggestion
    Where:
    
    - `<service-name>` is the name of the Service that you are applying the annotation to.
    - `<ingress-type` is the type of ingress that is used as the ingress class for the Service.
```

One last thing sorry, this was rendering weird cos there was no space :)

Reviewer: https://deploy-preview-4264--knative.netlify.app/development/developer/serving/services/ingress-class/#using-the-ingress-class-annotation

> # Configuring Services custom Ingress Class

Reviewer: ```suggestion
# Configuring Services custom ingress class
```

The `networking-config` ConfigMap is configured when Knative Serving is installed, and if no other ingress class is specified, this defaults to `"istio.ingress.networking.knative.dev"`. 

After it is configured in the `networking-config` ConfigMap, the `ingress.class` is used for all Knative Services that do not specify an `ingress.class` annotation.

!!! warning
    The ingress class value should only be modified during Knative Serving installation to avoid issues in your deployment.

## Using the ingress class annotation

Generally it is recommended for Knative Services to use the default `ingress.class`. However, in scenarios where there are multiple networking implementations, you might want to specify different ingress class annotations for each Service.

You can configure each Service to use a different ingress class by specifying the `network.knative.dev/ingress.class` annotation.

To add an ingress class annotation to a Service, run the following command:
    ```bash
    kubectl annotate kservice <service-name> network.knative.dev/ingress.class=<ingress-type>
    ```
    Where:
    
    - `<service-name>` is the name of the Service that you are applying the annotation to.

>     - `<ingress-type` is the type of ingress that is used as the ingress class for the Service.

Author: @abrennan89 sure.. I've added the indentation because it was like that in the suggestion, but it really looks odd.. so I am happy to remove that

>     - `<ingress-type` is the type of ingress that is used as the ingress class for the Service.

Author: @abrennan89 done.. sorry for the delay
